vendor/bat-native-ledger/src/bat/ledger/internal/server/api/endpoints/get_parameters_unittest.cc

```diff
@@ -0,0 +1,290 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/test/task_environment.h"
+#include "bat/ledger/internal/ledger_client_mock.h"
+#include "bat/ledger/internal/ledger_impl_mock.h"
+#include "bat/ledger/internal/server/api/endpoints/get_parameters.h"
+#include "bat/ledger/ledger.h"
+#include "net/http/http_status_code.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+// npm run test -- brave_unit_tests --filter=GetParametersTest.*
+
+using ::testing::_;
+using ::testing::Invoke;
+
+namespace ledger {
+namespace server {
+namespace api {
+
+class GetParametersTest : public testing::Test {
+ private:
+  base::test::TaskEnvironment scoped_task_environment_;
+
+ protected:
+  std::unique_ptr<ledger::MockLedgerClient> mock_ledger_client_;
+  std::unique_ptr<bat_ledger::MockLedgerImpl> mock_ledger_impl_;
+  std::unique_ptr<GetParameters> parameters_;
```

Comment:
A downside of heavily using namespaces is that we lose some important naming context when we bring some name into scope. In this case `GetParameters` is probably too vague. Something like `GetParametersEndpoint` or `GetApiParametersEndpoint` will be more self-documenting.

Note that if we went with `<something>Endpoint` then we don't really need to nest namespaces at all: `<root_namespace>::<something>Endpoint` works just fine.

Separately, I'm a little confused about our namespace strategy: should the root namespace for internal code be "ledger", "bat_ledger" or "braveledger"?

Reply:
`ledger` is what we are going with, everything else is legacy